pkg/kubectl/generate.go

```diff
@@ -35,17 +36,24 @@ type GeneratorParam struct {
 // Generator is an interface for things that can generate API objects from input parameters.
 type Generator interface {
 	// Generate creates an API object given a set of parameters
-	Generate(params map[string]string) (runtime.Object, error)
+	Generate(params map[string]interface{}) (runtime.Object, error)
```

Comment:
Looking at the way this has driven changes in usage, it might be useful to declare
```
type Params map[string]interface{}

func (p Params) GetString(key string) (string, error){}
func (p Params) GetBool(key string) (bool, error){}
func (p Params) GetStringSlice(key string) ([]string, error){}

```